staging/src/k8s.io/csi-translation-lib/plugins/azure_disk.go

```diff
@@ -58,6 +58,39 @@ func (t *azureDiskCSITranslator) TranslateInTreeStorageClassToCSI(sc *storage.St
 	return sc, nil
 }
 
+// TranslateInTreeInlineVolumeToCSI takes a Volume with AzureDisk set from in-tree
+// and converts the AzureDisk source to a CSIPersistentVolumeSource
+func (t *azureDiskCSITranslator) TranslateInTreeInlineVolumeToCSI(volume *v1.Volume) (*v1.PersistentVolume, error) {
+	if volume == nil || volume.AzureDisk == nil {
+		return nil, fmt.Errorf("volume is nil or Azure Disk not defined on volume")
+	}
+
+	azureSource := volume.AzureDisk
+	pv := &v1.PersistentVolume{
+		Spec: v1.PersistentVolumeSpec{
+			PersistentVolumeSource: v1.PersistentVolumeSource{
+				CSI: &v1.CSIPersistentVolumeSource{
+					Driver:           AzureDiskDriverName,
+					VolumeHandle:     azureSource.DataDiskURI,
+					ReadOnly:         *azureSource.ReadOnly,
+					FSType:           *azureSource.FSType,
+					VolumeAttributes: map[string]string{},
+				},
+			},
+			AccessModes: []v1.PersistentVolumeAccessMode{v1.ReadWriteOnce},
+		},
+	}
+
+	if *azureSource.CachingMode != "" {
```

Comment:
should this be `if azureSource != nil && *azureSource.CachingMode {`?

Reply:
good catch! I think we should just check for `if azureSource.CachingMode != nil` rather than something along the lines of `if azureSource.CachingMode != nil && *azureSource.CachingMode != ""` So:

```
        if azureSource.CachingMode != nil {
                pv.Spec.PersistentVolumeSource.CSI.VolumeAttributes[azureDiskCachingMode] = string(*azureSource.CachingMode)
        }
```

Seems like this is something applicable to the above code as well for the regular `TranslateInTreePVToCSI` already there for AzureDisk at https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/csi-translation-lib/plugins/azure_disk.go#L79. Will file an issue to fix it it both places.